features/timeseries/src/main/java/org/opennms/netmgt/timeseries/samplewrite/RingBufferTimeseriesWriter.java

```diff
@@ -154,55 +161,102 @@ private void setUpWorkerPool() {
         ringBuffer.addGatingSequences(workerPool.getWorkerSequences());
 
         workerPool.start(executor);
+
+        readyToRockAndRoll.set(true);
     }
 
     @Override
     public void destroy() {
+        readyToRockAndRoll.set(false);
```

Comment:
Do we need to reset this back to true ?

Reply:
It is only set to `true` in `setUpWorkerPool`, which is called from the constructor. Is it reasonable from a Kafka standpoint to require a new instance of the class to be constructed after `destroy` has been called?